src/sagemaker/workflow/entities.py

```diff
@@ -57,3 +58,68 @@ class Expression(abc.ABC):
     @abc.abstractmethod
     def expr(self) -> RequestType:
         """Get the expression structure for workflow service calls."""
+
+
+class PipelineVariable(Expression):
+    """Base object for pipeline variables
+
+    PipelineVariables must implement the expr property.
+    """
+
+    def __add__(self, other: Union[Expression, PrimitiveType]):
+        """Add function for PipelineVariable
+
+        Args:
+            other (Union[Expression, PrimitiveType]): The other object to be added.
+
+        Return:
+            `sagemaker.workflow.functions.Join`
+        """
+        from sagemaker.workflow.functions import Join
+
+        if isinstance(self, Join):
+            self.values.append(other)  # pylint: disable=no-member
+            return self
+
+        return Join(on="", values=[self, other])
+
+    def __str__(self):
+        """String function for PipelineVariable"""
+        return json.dumps(self._expr_helper(True))
```

Comment:
> I don't think we should do json dump here. We do json dump at very end of the pipeline definition compilation. It should return a pipeline variable of string type instead of a real Python string.

Same above